student/src/components/Form/index.js

```diff
@@ -24,13 +25,21 @@ class MainForm extends Component {
       return <option value={o.value}> {o.label}</option>;
     });
   }
-  generateRadioOptions(options) {
+
+  onRadioClick = (name, value) => {
+    this.setState((prevState) => {
+      let newRadios = prevState.radios;
+      newRadios[name] = value;
+      return { radios: newRadios };
+    });
```

Comment:
For code modulration, and best practice, see the following code 
```
  onRadioClick = (name, value) => {
    console.log(name);
    this.setState(
      prevState => {
        return { radios: { ...prevState.radios, [name]: value } };
      },
      () => {
        console.log(this.state);
      },
    );
  };
```
So `this.setState` could take a secound parameter which is a callback that will be called after the new state is set, instead of using `setTimeout`